lightning/src/routing/network_graph.rs

```diff
@@ -1123,6 +1133,64 @@ impl NetworkGraph {
 		self.update_channel_from_unsigned_announcement_intern(msg, None, chain_access)
 	}
 
+	/// Update channel from partial announcement data received via rapid gossip
+	///
+	/// `timestamp: u64`: Timestamp the announcement was originally received.
+	///
+	/// All other parameters as used in [`msgs::UnsignedChannelAnnouncement`] fields.
+	pub fn update_channel_from_partial_announcement(&self, short_channel_id: u64, timestamp: u64, features: ChannelFeatures, node_id_1: PublicKey, node_id_2: PublicKey) -> Result<(), LightningError> {
+		if node_id_1 == node_id_2 {
+			return Err(LightningError{err: "Channel announcement node had a channel with itself".to_owned(), action: ErrorAction::IgnoreError});
+		};
+
+		let node_1 = NodeId::from_pubkey(&node_id_1);
+		let node_2 = NodeId::from_pubkey(&node_id_2);
+		let chan_info = ChannelInfo {
+			features,
+			node_one: node_1.clone(),
+			one_to_two: None,
+			node_two: node_2.clone(),
+			two_to_one: None,
+			capacity_sats: None,
+			announcement_message: None,
+			announcement_received_time: timestamp,
+		};
+
+		let mut channels = self.channels.write().unwrap();
+		let nodes = self.nodes.write().unwrap();
+		match channels.entry(short_channel_id) {
```

Comment:
Should we inline this in `add_channel_between_nodes` as well, with the `utxo_value` field from the bottom implementation passed in and always `None` when called from `update_channel_from_partial_announcement`?

Reply:
we might? It will change the signature of add channels between nodes and require it to return a result.

Reply:
Actually, no, because the channel tree logic here is different from the one that's required by the other call site in the `update_channel_from_unsigned_announcement_intern` function. We could add a flag to indicate not to remove a channel between nodes, but at that point the functionality gets confusing for the reader.